cmd/kpod/push.go

```diff
@@ -120,7 +137,25 @@ func pushCmd(c *cli.Context) error {
 			RemoveSignatures: removeSignatures,
 			SignBy:           signBy,
 		},
+		AuthFile: c.String("authfile"),
 	}
 
 	return runtime.PushImage(srcName, destName, options, writer)
 }
+
+// get the registry from the destination path
+func getRegistryFromPath(image string) (string, error) {
+	if !strings.Contains(image, "docker://") {
+		return "", nil
+	}
+	image = image[9:]
+	imgRef, err := reference.Parse(image)
+	if err != nil {
+		return "", err
+	}
+	registry := reference.Domain(imgRef.(reference.Named))
+	if registry == "" {
+		registry = "docker.io"
```

Comment:
Using `reference.ParseNormalizedNamed` would avoid having to hard-code this domain name here.  Or, preferably? use `transports.ParseImageName`+`DockerReference()`.